servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/CompletableMergeWithPublisher.java

```diff
@@ -67,7 +200,6 @@ void handleSubscribe(final Subscriber<? super T> subscriber, final SignalOffload
             // invoke it from the mergeWith Publisher Executor thread.
             this.offloadedSubscriber = new ConcurrentTerminalSubscriber<>(signalOffloader.offloadSubscriber(
```

Comment:
We use `ConcurrentSubscription` and `ConcurrentTerminalSubscriber` to handle concurrency in this class. The `ConcurrentSubscription` has a comment clarifying why it's required. Consider adding a similar comment for `ConcurrentTerminalSubscriber`.

Reply:
`ConcurrentTerminalSubscriber` usage isn't changed in this PR and it is removed all together in #1334 so lets defer on this for now.